### PR TITLE
Flash: Retry failed payment attempts

### PR DESCRIPTION
**Pull request by omerfirmak**
See each commit

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
I will have a closer look after it's out of draft. You think we can merge https://github.com/bpfkorea/agora/pull/1721? I'll rebase against v0.x.x after.

**Comment by omerfirmak** [comment]
Down the road we should probably add a max retry count and an increasing delay between each attempt.

File: source/agora/test/Flash.d [hunk 1]
```diff
@@ -419,3 +419,120 @@ unittest
     writefln("alice closing tx: %s", alice.getClosingTx(alice_bob_chan_id));
     assert(block12.txs[0] == alice.getClosingTx(alice_bob_chan_id));
 }
+
+/// Test indirect channel payments
```
Comment (AndrejMitrovic) [inline source/agora/test/Flash.d]: Should the comment be updates? Seems like copypasta. :)

I'll merge now and you can fix it.later.

**Review by AndrejMitrovic — APPROVED** [review]

File: source/agora/test/Flash.d (hunk 1 above)
Comment (AndrejMitrovic) [reply source/agora/test/Flash.d]: Or self-merge when ready!

Comment (omerfirmak) [reply source/agora/test/Flash.d]: Thanks, updated.

**Pull request merged**